classes/template/PKPTemplateManager.inc.php

```diff
@@ -641,6 +641,28 @@ function registerJSLibrary() {
 			)
 		);
 
+		// Load constants for new component library
+		$const = array(
```

Comment:
I think all this can be removed if you use `define_exposed` where the roles are first defined, rather than `define`.

Reply:
I know that this is a bit messy, but I'm slowly trying to wean us off of the `$.pkp` namespace used in the main JS lib and towards a `pkp` global we're using with the new Vue.js system. A couple of reasons:

1. As we replace more of the old framework, I'll eventually be able to get rid of `$.pkp`. Until then, it's really hard to track dependencies for the data inside of it. The new approach is for components to register their own constant dependents (eg - [SubmissionListHandler](https://github.com/pkp/pkp-lib/blob/master/controllers/list/submissions/PKPSubmissionsListHandler.inc.php#L152-L176)) and these are fed into the global `pkp.const`. This approach aligns with the new `pkp.const.ANY_CONSTANT` way of retrieving constants from components.

2. With the style guide that we're building alongside the new Vue.js components, we have to [manually set up constants](https://github.com/pkp/ui-library/blob/master/src/main.js#L20-L47). It helps for that app to have these in `pkp.const`.

Reply:
Gotcha, that makes sense. I remember talking about this before.